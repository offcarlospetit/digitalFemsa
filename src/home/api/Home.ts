// Need to use the React-specific entry point to import createApi
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { Product } from '../types';

// Define a service using a base URL and expected endpoints
export const productApi = createApi({
    reducerPath: 'prductsApi',
    baseQuery: fetchBaseQuery({
        baseUrl: 'https://6222994f666291106a29f999.mockapi.io/api/v1/',
    }),

    tagTypes: ['Products'],
    endpoints: (builder) => ({
        getProducts: builder.query({
            query: () => 'products',
            providesTags: ['Products'],
            transformResponse: (response: Product[]) => response
        }),
    }),

});

// Export hooks for usage in functional components, which are
// auto-generated based on the defined endpoints
export default productApi;