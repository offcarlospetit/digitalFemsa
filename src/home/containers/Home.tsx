import React from 'react';
import {Box, Text} from '../../uikit';

type Props = {};

const Home: React.FC<Props> = ({}) => {
  return (
    <Box flex={1} justifyContent="center">
      <Text>Home</Text>
    </Box>
  );
};

export default Home;
