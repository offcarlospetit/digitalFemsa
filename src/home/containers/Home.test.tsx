import * as React from 'react';
import Home from './Home';
import {render, waitFor} from '../../../test/test-utils';
import {rest} from 'msw';
import {setupServer} from 'msw/node';

const createTestProps = (props: Object): any => ({
  navigation: {
    navigate: jest.fn(),
    dispatch: jest.fn(),
  },
  ...props,
});

export const handlers = [
  rest.get('products', (req, res, ctx) => {
    return res(ctx.json('John Smith'), ctx.delay(150));
  }),
];

const server = setupServer(...handlers);

describe('Home Component', () => {
  // Enable API mocking before tests.
  beforeAll(() => server.listen());

  // Reset any runtime request handlers we may add during the tests.
  afterEach(() => server.resetHandlers());

  // Disable API mocking after the tests are done.
  afterAll(() => server.close());

  test(`Should renders Home Component correctly `, async () => {
    const props = createTestProps({});
    const snap = await render(<Home {...props} />).toJSON();
    await waitFor(() => expect(snap).toMatchSnapshot());
  });
});
