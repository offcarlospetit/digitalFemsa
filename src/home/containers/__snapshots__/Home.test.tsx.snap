// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should renders Home Component correctly  1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f8f8f8",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "paddingLeft": 20,
          "paddingTop": 10,
        },
        {
          "marginTop": 13,
        },
      ]
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0B0B0B",
              "fontSize": 20,
              "fontWeight": "800",
              "lineHeight": 27.32,
            },
          ]
        }
      >
        Bienvenido de vuelta!
      </Text>
      <Text
        style={
          [
            {
              "color": "#0B0B0B",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 21.86,
            },
          ]
        }
      >
        Ruben Rodriguez
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#9B9898",
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
          },
        ]
      }
    >
      TUS PUNTOS
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 40,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 24,
            "paddingTop": 20,
          },
          [
            {
              "backgroundColor": "#334FFA",
              "borderRadius": 20,
              "elevation": 5,
              "height": 143,
              "padding": 10,
              "shadowColor": "#0B0B0B",
              "shadowOffset": {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 4,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "800",
              "lineHeight": 21.86,
              "marginLeft": 20,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontWeight": "800",
              "lineHeight": 43.71,
              "marginTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        0.00 pts
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#9B9898",
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
          },
        ]
      }
    >
      TUS MOVIMIENTOS
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 24,
            "paddingTop": 20,
          },
          [
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "height": 350,
              "padding": 10,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 24,
          "paddingHorizontal": 24,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#334FFA",
            "borderRadius": 10,
            "minWidth": 170,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingHorizontal": 24,
            "paddingTop": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "800",
                "lineHeight": 21.86,
              },
            ]
          }
        >
          Ganados
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#334FFA",
            "borderRadius": 10,
            "minWidth": 170,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingHorizontal": 24,
            "paddingTop": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "800",
                "lineHeight": 21.86,
              },
            ]
          }
        >
          Canjeados
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Item Component Should renders  1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f8f8f8",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "paddingLeft": 20,
          "paddingTop": 10,
        },
        {
          "marginTop": 13,
        },
      ]
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0B0B0B",
              "fontSize": 20,
              "fontWeight": "800",
              "lineHeight": 27.32,
            },
          ]
        }
      >
        Bienvenido de vuelta!
      </Text>
      <Text
        style={
          [
            {
              "color": "#0B0B0B",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 21.86,
            },
          ]
        }
      >
        Ruben Rodriguez
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#9B9898",
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
          },
        ]
      }
    >
      TUS PUNTOS
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 40,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 24,
            "paddingTop": 20,
          },
          [
            {
              "backgroundColor": "#334FFA",
              "borderRadius": 20,
              "elevation": 5,
              "height": 143,
              "padding": 10,
              "shadowColor": "#0B0B0B",
              "shadowOffset": {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 4,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "800",
              "lineHeight": 21.86,
              "marginLeft": 20,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontWeight": "800",
              "lineHeight": 43.71,
              "marginTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        0.00 pts
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#9B9898",
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
          },
        ]
      }
    >
      TUS MOVIMIENTOS
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 20,
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 24,
            "paddingTop": 20,
          },
          [
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "height": 350,
              "padding": 10,
              "width": "100%",
            },
          ],
        ]
      }
    >
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 24,
          "paddingHorizontal": 24,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#334FFA",
            "borderRadius": 10,
            "minWidth": 170,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingHorizontal": 24,
            "paddingTop": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "800",
                "lineHeight": 21.86,
              },
            ]
          }
        >
          Ganados
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#334FFA",
            "borderRadius": 10,
            "minWidth": 170,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingHorizontal": 24,
            "paddingTop": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "800",
                "lineHeight": 21.86,
              },
            ]
          }
        >
          Canjeados
        </Text>
      </View>
    </View>
  </View>
</View>
`;
